src/groups.py

```diff
@@ -20,9 +20,11 @@ def draw(self, target_pos: Coordinate):
 
         for layer in LAYERS.values():
 
-            for sprite in sorted(
-                    self.sprites(),
-                    key=lambda spr: spr.rect.centery):
+            if layer == LAYERS["main"]:
+                sorted_sprites = sorted(self.sprites(), key=lambda sprite: sprite.hitbox_rect.centery)
+            else:
+                sorted_sprites = sorted(self.sprites(), key=lambda sprite: sprite.rect.centery)
+
+            for sprite in sorted_sprites:
                 if sprite.z == layer:
-                    self.display_surface.blit(
-                        sprite.image, sprite.rect.topleft + (self.offset if not isinstance(sprite, TextBox) else (0, 0)))
+                    sprite.draw(self.display_surface, self.offset)
```

Comment:
This change makes it so text boxes do not get displayed on the screen regardless of the player’s position anymore.

Reply:
i need to create draw function in textBoxes good catch

Reply:
…or you could just revert this change

Reply:
he thing is, the text boxes are not part of the AllSprites group, so I don’t understand why those lines were there. It’s never going to display the text boxes anyway since they are not part of it; they are part of a list _tb_list.

Reply:
Actually, they are. Which is why these lines are there

Reply:
Oh yes, sorry, I just saw that.

So my question is, why doesn’t the TextBox class inherit from the Sprite class (not the pygame one)?

Normally, it should behave the same as with the isinstance check with the draw function I added in TextBox. How do I check if they work because, even before, I never saw them?

Reply:
The non-pygame Sprite base class requires parameters that are useless in the context of displaying textboxes

Reply:
okok